frontend/src/plugins/persistStore.ts

```diff
@@ -48,30 +50,40 @@ export function persistStorePlugin(context: PiniaPluginContext): void {
               .delete()
             break
           case 'openFile':
-            localStorage.setItem('mainStore.currentContractId', args[0] as string)
-            localStorage.setItem('mainStore.openedFiles', store.openedFiles.join(','))
+            localStorage.setItem('contractsStore.currentContractId', args[0] as string)
+            localStorage.setItem('contractsStore.openedFiles', store.openedFiles.join(','))
             break
           case 'closeFile':
-            localStorage.setItem('mainStore.currentContractId', store.currentContractId)
-            localStorage.setItem('mainStore.openedFiles', store.openedFiles.join(','))
+            localStorage.setItem('contractsStore.currentContractId', store.currentContractId)
+            localStorage.setItem('contractsStore.openedFiles', store.openedFiles.join(','))
             break
           case 'addDeployedContract':
             await upsertDeployedContract(args[0] as DeployedContract)
             break
           case 'setCurrentContractId':
-            localStorage.setItem('mainStore.currentContractId', args[0] as string)
+            localStorage.setItem('contractsStore.currentContractId', args[0] as string)
             break
-          case 'generateNewAccount':
-            localStorage.setItem('mainStore.accounts', store.accounts.join(','))
-            localStorage.setItem('mainStore.currentUserAddress', store.currentUserAddress)
+
+          case 'removeDeployedContract':
+            await db.deployedContracts
+              .where('contractId')
+              .equals(args[0] as string)
+              .delete()
             break
-            case 'removeDeployedContract':  
-            await db.deployedContracts.where('contractId').equals(args[0] as string).delete()
+          default:
+            break
+        }
+      } else if (store.$id === 'accountsStore') {
```

Comment:
@bradleySuira this separation is a good first step, but I would continue separating these methods on different files and merging all of them on this one. Not for now though